packages/database/src/prisma-api.ts

```diff
@@ -409,22 +409,12 @@ export function getSdk(
         },
       })
 
-      // TODO: This doesn't seem to be looking up the bulk coupon based
-      // on the product ID which will become an issue when any particular
-      // app has more than one product.
-      // E.g. should probably account for `restrictedToProductId`
-      //
-      // Check if this user has already purchased a bulk coupon, in which
-      // case, we'll be able to treat this purchase as adding seats.
-      //
-      // TODO: I believe the `maxUses` check is redundant. If there is at
-      // least one `bulkCouponPurchase` attached to this Coupon, then it is a
-      // bulk coupon for this user.
+      // Check if this user has already purchased a bulk coupon for this
+      // product, in which case, we'll be able to treat this purchase as
+      // adding seats.
       const existingBulkCoupon = await ctx.prisma.coupon.findFirst({
         where: {
-          maxUses: {
-            gt: 1,
-          },
+          restrictedToProductId: productId,
```

Comment:
For apps that will eventually need to support multiple products, we'll need this `restrictedToProductId` filter in here so that we are looking up the correct bulk coupon.